icicle/include/api/babybear.h

```diff
@@ -32,47 +32,49 @@ extern "C" cudaError_t babybear_create_poseidon2_cuda(
   device_context::DeviceContext& ctx
 );
 
-extern "C" cudaError_t babybear_init_poseidon2_cuda(
+extern "C" cudaError_t babybear_poseidon2_load_cuda(
   poseidon2::Poseidon2<babybear::scalar_t>** poseidon,
   unsigned int width,
   poseidon2::MdsType mds_type,
   poseidon2::DiffusionStrategy diffusion,
   device_context::DeviceContext& ctx
 );
 
-extern "C" cudaError_t babybear_poseidon2_permute_many_cuda(
+extern "C" cudaError_t babybear_poseidon2_absorb_many_cuda(
+  const poseidon2::Poseidon2<babybear::scalar_t>* poseidon,
+  const babybear::scalar_t* inputs,
+  babybear::scalar_t* states,
+  unsigned int number_of_states,
+  unsigned int input_block_len,
+  hash::SpongeConfig& cfg);
```

Comment:
probably better to make it const reference assuming you are not going to modify it inside

Reply:
I'll leave it like that for now, because in all of our primitives we use non-const configs. We can change them all at once in the future